[newrelic-logging] Update namespace and cluster variables

bpschmitt opened the pull request:
#### Is this a new chart

No

#### What this PR does / why we need it:

- Modifies `namespace` for the manifests to `newrelic` from `default`
- Uses the release `namespace` instead of always installing into `default`
- Standardizes on `cluster` instead of `clusterName` to be consistent with `nri-bundle`

#### Which issue this PR fixes

N/A

#### Special notes for your reviewer:

This is a cleaner version of #185 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[mychartname]`)

bpschmitt commented:
@jsubirat Please review ASAP.  Also, please look into https://github.com/newrelic/newrelic-fluent-bit-output/issues/62.

jsubirat reviewed (changes requested):
Hey @bpschmitt , left a few comments. Some additional questions:
- Can you please include the `cluster` and `global.cluster` parameters to the chart README.md? Similar to [here](https://github.com/newrelic/helm-charts/tree/master/charts/newrelic-infrastructure#configuration).
- So, for now, we'll keep the current behavior of still sending `cluster_name` as a log record attribute even if it's not specified, right? I'm totally up for not fixing it in this PR, I just want to double-check.

jsubirat commented on charts/newrelic-logging/k8s/new-relic-fluent-plugin.yml:
```diff
@@ -2,7 +2,7 @@ apiVersion: apps/v1
+  namespace: newrelic
   labels:
     k8s-app: newrelic-logging
     version: v1
```
Please note this applies to both the K8s plugin and the Helm chart template.
- For the K8s plugin, I'd suggest to leave that line with just a `*` wildcard to avoid having to update it for each new version.
- For the Helm chart template, you can also use just the wildcard, or parametrize it with `{{ .Chart.AppVersion }}`. In this case, I think it makes sense to use the latter option.

bpschmitt commented:
@jsubirat please take another look when you get some time.

Latest updates include:

- Bumped `appVersion` to `1.4.6`
- Reverting to static `out_newrelic.so` file based upon our conversation earlier
- Removed the `namespace.yml` and reverted `README` to include the namespace creation command

jsubirat commented on charts/newrelic-logging/k8s/new-relic-fluent-plugin.yml:
```diff
@@ -41,7 +41,7 @@ spec:
             - name: PATH
               value: "/var/log/containers/*.log"
-          image: newrelic/newrelic-fluentbit-output:1.4.3
+          image: newrelic/newrelic-fluentbit-output:1.4.5
```
Hey @bpschmitt , please update this to `1.4.6` as well.

jsubirat reviewed (changes requested):
Thanks for the changes @bpschmitt . They look good to me, just make sure to update the version in the K8s folder as well and you're good to go.

jsubirat reviewed (approved):
LGTM. 🎉  Thanks a lot for your patience and dedication in delivering this piece of excellent work, @bpschmitt ! 🙇
